daemon: factor out node config headerfile into separate function

ianvernon opened the pull request:
This makes the code more readable / modular.

Signed-off by: Ian Vernon <ian@cilium.io>

Commit:
daemon: factor out node config headerfile into separate function

This makes the code more readable / modular.

Signed-off by: Ian Vernon <ian@cilium.io>

aanm reviewed (commented):
Cool! `createNodeConfigHeaderfile` only needs the `d.loopbackIPv4` can we put the `d.lookbackIPv4` in the `pkg/node` and move this whole function to elsewhere like `pkg/bpf`?

Commit:
pkg/node: move IPv4Loopback address from daemon to node package

Signed-off by: Ian Vernon <ian@cilium.io>

aanm commented on daemon/daemon.go:
```diff
@@ -1145,8 +1157,8 @@ func NewDaemon() (*Daemon, error) {
 		}
-		d.loopbackIPv4 = loopbackIPv4
-		log.Infof("  Loopback IPv4: %s", d.loopbackIPv4.String())
+		node.SetIPv4Loopback(loopbackIPv4)
```
`d.loopbackIPv4` can be removed now

ianvernon replied:
Done.

Commit:
daemon: remove loopbackIPv4 from Daemon type

This value is now contained within pkg/node.

Signed-off by: Ian Vernon <ian@cilium.io>
